# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1228903468 -3600
#      Wed Dec 10 11:04:28 2008 +0100
# Node ID ff5892c8ddd07f72d8f579ca45ff65d2a6e1909d
# Parent  3cbe3054fa7e34abcec4133bcf0343ef7142cfd7
fix integer assignment to non-existing variables

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-12-10  Jaroslav Hajek  <highegg@gmail.com>
+
+	* ov-intx.h (OCTAVE_VALUE_INT_SCALAR_T::empty_clone): Construct an
+	empty matrix instead of zero.
+
 2008-12-07  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
         * mappers.cc (Ftolower): Make lower alias of tolower, add tests
         * mappers.cc (Ftoupper): Make upper alias of toupper, add tests
 
 2008-12-09  Jaroslav Hajek  <highegg@gmail.com>
 
 	* DLD-FUNCTIONS/balance.cc (Fbalance): Exploit the new AEPBAL functionality.
diff --git a/src/ov-intx.h b/src/ov-intx.h
--- a/src/ov-intx.h
+++ b/src/ov-intx.h
@@ -397,17 +397,17 @@ public:
     : octave_base_int_scalar<OCTAVE_INT_T> (nda) { }
 
   ~OCTAVE_VALUE_INT_SCALAR_T (void) { }
 
   octave_base_value *clone (void) const
     { return new OCTAVE_VALUE_INT_SCALAR_T (*this); }
 
   octave_base_value *empty_clone (void) const
-    { return new OCTAVE_VALUE_INT_SCALAR_T (); }
+    { return new OCTAVE_VALUE_INT_MATRIX_T (); }
 
   octave_value do_index_op (const octave_value_list& idx,
 			    bool resize_ok = false)
     {
       octave_value retval;
 
       if (idx.valid_scalar_indices ())
 	retval = scalar;
